src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -153,16 +153,6 @@ public static int Main(string[] args)
                 log.Error("Exception in RemoteTestAgent", ex);
             }
 
-            try
-            {
-                // Unregister the channel
-                testAgencyServer.Stop();
```

Comment:
@CharliePoole I don't follow. Are you saying that `TestAgency` should not be a `ServerBase`?

Any `ServerBase` that needs remoting will be susceptible to the SocketException any time the remote instance of the `ServerBase` is shutting down, so unless I 'm missing something, `ServerBase` is definitely where the fix belongs.